Utils/ScriptValidator/Unit1.pas

```diff
@@ -53,45 +73,128 @@ procedure TForm1.FormCreate(Sender: TObject);
 procedure TForm1.FormDestroy(Sender: TObject);
 begin
   fScripting.Free;
+  sListFileInFolder.Free;
   DragAcceptFiles(Handle, False);
 end;
 
-procedure TForm1.btnBrowseClick(Sender: TObject);
+procedure TForm1.btnBrowseFileClick(Sender: TObject);
 begin
   if OpenDialog1.Execute then
-    Edit1.Text := OpenDialog1.FileName;
+  Edit1.Text := OpenDialog1.FileName;
 end;
 
 
+procedure TForm1.btnBrowsePathClick(Sender: TObject);
+var DirValidate : String;
+begin
+  if not SelectDirectory('Select the folder to Validate scripts', '', DirValidate) then Exit;
+  Edit1.Text := DirValidate;
+end;
+
+procedure TForm1.Edit1Change(Sender: TObject);
+begin
+  If (GetFileAttributes(PWideCHar(TEdit(Sender).Text)) and FILE_ATTRIBUTE_DIRECTORY) = FILE_ATTRIBUTE_DIRECTORY then
+  begin
+    IsValidatePath := true;
+    Memo1.Text := 'Selected folders!';
+  end
+  else
+  begin
+    IsValidatePath := false;
+    Memo1.Text := 'Select Files!';

```

Comment:
Why is the text different, there you reference selectED folders, and here select files?